src/components/Input/Input.tsx

```diff
@@ -1,12 +1,26 @@
 import React, { forwardRef } from 'react';
-import PropTypes from 'prop-types';
 import bem from '../../utils/bem';
 import styles from './Input.scss';
-import { CONTEXTS, INPUT_TYPES, SIZES } from '../../constants';
+import { ContextTypes, InputTypes, SizesTypes } from '../../constants';
+
+// Any other attributes (onChange, onKeyUp etc.) are
+// supported although not defined in props type definition
+interface Props extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size'> {
```

Comment:
At our meeting we said that as part of this ticket there will be some document as to say what the developer guidelines are. Like when to use `interface` and when `types`. Could you address it?

Reply:
That was not my understanding from the meeting. If we have no guidelines on things that are easy to write in different ways, we will end up with a mess pretty soon. It is not even sure then if this is something we will need to change right after it is submitted. I don't mean we have to have a comprehensive guide to TS, only some basic standards that we already run into, such as using type or interface and when.

Reply:
For `interfaces vs types` I'd suggest to stick with interfaces, cause `interface` allows to extend Props from 3rd party libs and it can be useful sometimes (as we see above). So for the sake of consistency `interface` would be nicer from my pov.

Reply:
Sounds ok to me. Honestly, I had no time to look into the differences between them, but I see that this PR already uses both. So it would be good to know when which.

Reply:
Hmm, I'd rather go with one option for consistency and less time spent on choosing what to use for a particular cause

Reply:
Maybe you can discuss it with Alexey and come up with a convention. Currently src/constants/style-related.ts uses `type`.